launch/launch/launch_description_source.py

```diff
@@ -51,21 +70,43 @@ def try_get_launch_description_without_context(self) -> Optional[LaunchDescripti
         This method is useful for trying to introspect the included launch
         description without visiting the user of this source.
         """
+        if self.__launch_description is None:
+            # Try to expand the launch file path and load the launch file with a local context.
+            try:
+                context = LaunchContext()
+                expanded_location = \
+                    perform_substitutions(context, self.__location)
+                return self.__get_method(expanded_location)
+            except Exception as exc:
+                self.__logger.debug(traceback.format_exc())
+                self.__logger.debug(
+                    'Failed to load the launch file without a context: ' + str(exc),
+                )
         return self.__launch_description
 
     def get_launch_description(self, context: LaunchContext) -> LaunchDescription:
         """Get the LaunchDescription, loading it if necessary."""
+        if self.__expanded_location is None:
+            self.__expanded_location = \
+                perform_substitutions(context, self.__location)
         if self.__launch_description is None:
-            raise RuntimeError(
-                'LaunchDescriptionSource.get_launch_description(): '
-                'called without launch description being set'
-            )
+            self.__launch_description = \
+                self.__get_method(self.__expanded_location)
         return self.__launch_description
 
     @property
     def location(self) -> str:
-        """Getter for self.__location."""
-        return self.__location
+        """
+        Get the location of the Python launch file as a string.
```

Comment:
@ivanpauno consider:
```suggestion
        Get the location of the launch description as a string.
```